rcl/src/rcl/arguments.c

```diff
@@ -1074,6 +1162,77 @@ _rcl_parse_param_file_rule(
   return RCL_RET_INVALID_PARAM_RULE;
 }
 
+rcl_ret_t
+_rcl_parse_external_log_config_file(
+  const char * arg,
+  rcl_allocator_t allocator,
+  char ** log_config_file)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
+
+  const size_t param_prefix_len = sizeof(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE);
+  if (strncmp(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE, arg, param_prefix_len) == 0) {
+    size_t outlen = strlen(arg) - param_prefix_len;
+    log_config_file = allocator.allocate(sizeof(char) * (outlen + 1), allocator.state);
+    if (NULL == log_config_file) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR("Failed to allocate memory for parameters file path\n");
+      return RCL_RET_BAD_ALLOC;
+    }
+    snprintf(*log_config_file, outlen + 1, "%s", arg + param_prefix_len);
+    return RCL_RET_OK;
+  }
+
+  RCL_SET_ERROR_MSG("Argument does not start with '" RCL_EXTERNAL_LOG_CONFIG_ARG_RULE "'");
+  return RCL_RET_INVALID_PARAM_RULE;
+}
+
+RCL_LOCAL
+rcl_ret_t
+_rcl_parse_bool_arg(
+  const char * arg,
+  const char * key,
+  bool * value)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(key, RCL_RET_INVALID_ARGUMENT);
+
+  const size_t param_prefix_len = strlen(key);
+  if (strncmp(key, arg, param_prefix_len) == 0) {
+    return _atob(arg + param_prefix_len, value);
```

Comment:
Is value allowed to be `NULL`? If not should it be checked on the input at the beginning of the function.

Reply:
Oh I guess it's checked in `_atob` anyways.